lib/ansible/modules/storage/purestorage/purefa_host.py

```diff
@@ -173,9 +200,12 @@ def main():
         iqn=dict(type='list'),
         wwns=dict(type='list'),
         volume=dict(type='str'),
+        personality=dict(type='str', default='',
+                         choices=['hpux', 'vms', 'aix', 'esxi', 'solaris',
+                                  'hitachi-vsp', 'oracle-vm-server', '']),
     ))
 
-    module = AnsibleModule(argument_spec, supports_check_mode=True)
+    module = AnsibleModule(argument_spec, supports_check_mode=False)
```

Comment:
Why has this changed?
What's needed to support checkmode again?

Reply:
when doing this update I realized that my original code said it supported check mode, but in reality, it didn't, hence the change. All the Pure modules don't support check mode. When I first wrote these I didn't understand what check mode really was so just assumed we supported it.

Reply:
check_mode was explained to me by someone else in  another PR and I was told not to have it set as true due to the structure of my modules